guidance/library/_json.py

```diff
@@ -821,74 +1167,53 @@ def json(
         lm,
         *,
         json_schema: JSONSchema,
+        base_uri: str,
     ):
         if json_schema is True:
             json_schema = {}
         elif json_schema is False:
-            raise ValueError("No valid JSON can be generated from a schema of `False`")
+            raise UnsatisfiableSchemaError("No valid JSON can be generated from a schema of `false`")
 
         if json_schema == {}:
             return lm + self.any()
 
-        validate_json_node_keys(json_schema)
+        # Early exit for simple cases
+        if Keyword.CONST in json_schema:
+            return lm + self.const(value=json_schema[Keyword.CONST], parent_schema=json_schema)
 
-        if Keyword.ANYOF in json_schema:
-            sibling_keys = get_sibling_keys(json_schema, Keyword.ANYOF)
-            if sibling_keys:
-                raise NotImplementedError(f"anyOf with sibling keys is not yet supported. Got {sibling_keys}")
-            return lm + self.anyOf(anyof_list=json_schema[Keyword.ANYOF])
+        if Keyword.ENUM in json_schema:
+            return lm + self.enum(options=json_schema[Keyword.ENUM], parent_schema=json_schema)
+
+        # More complex cases; validation needed
+        validate_json_node_keys(json_schema)
+        json_schema = self.push_sibling_keys(json_schema)
 
         if Keyword.ALLOF in json_schema:
             sibling_keys = get_sibling_keys(json_schema, Keyword.ALLOF)
-            if sibling_keys:
-                raise NotImplementedError(f"allOf with sibling keys is not yet supported. Got {sibling_keys}")
-            allof_list = json_schema[Keyword.ALLOF]
-            if len(allof_list) != 1:
-                raise ValueError("Only support allOf with exactly one item")
-            return lm + self.json(json_schema=allof_list[0])
+            assert not sibling_keys
```

Comment:
I assume that these bare assertions (without even a text message) should never be hit, unless things are going _really_ wrong?

Reply:
Honestly, there's a philosophy that "production code" should never have bare asserts, which I may agree with. But yes, these asserts are essentially "whoever wrote this code really messed up".

Reply:
Sounds like it's good to leave them in as-is. I tend to like rechecking things too, since it's often the really simple bit of code that couldn't possibly be wrong.....